lib/providers/base/base_provider.dart

```diff
@@ -101,15 +103,17 @@ abstract class BaseProvider extends ChangeNotifier {
     }
   }
 
-  Future<QueryResult> runMutation({
-    required DocumentNode document,
-    Map<String, dynamic>? variables,
-    bool logFailures = true,
-  }) async {
+  Future<QueryResult> runMutation(
+      {required DocumentNode document,
+      Map<String, dynamic>? variables,
+      bool logFailures = true,
+      FutureOr<void> Function(GraphQLDataProxy, QueryResult<Object?>?)?
+          update}) async {
```

Comment:
Nit: if you put a comma between `update` and `}` then it will be nicely formatted:
```dart
Future<QueryResult> runMutation({
  required DocumentNode document,
  Map<String, dynamic>? variables,
  bool logFailures = true,
  FutureOr<void> Function(GraphQLDataProxy, QueryResult<Object?>?)? update,
}) async {
```